src/algorithms.cpp

```diff
@@ -22,4 +24,105 @@ bool isComplete(const Model &model) {
     return true;
 }
 
+bool isLocCorrect(const Model& model, IdRef id){
```

Comment:
1. I'm not sure this is the best name for this function. How about `isLocallyCorrect`?
2. Since this function is not meant to be linked against externally, wrap it in an anonymous namespace

Reply:
1. Renamed in 537a6a5a98177040cdd047cf470162546280aa7d
2. I think it would be great to test `isLocallyCorrect` separately from `isCorrect`, as this will require example model to only be partially correct. For this purposes created a new private header (537a6a5a98177040cdd047cf470162546280aa7d). As an example added a test for `isLocallyCorrect` (0ccf8cd6827c7677c52f30222378d123d5517309). More tests pending.